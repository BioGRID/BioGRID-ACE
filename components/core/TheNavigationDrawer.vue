<template>
    <div class="the-navigation-drawer">
        <v-navigation-drawer
            v-model="drawerState"
            :mini-variant="miniVariant"
            :clipped="clipped"
            :disable-resize-watcher="true"
            color="primary"
            width="200"
            fixed
            app
        >
            <v-list-item class="mt-1">
                <v-list-item-title v-if="miniVariant" class="title ml-n3 mr-n3 mt-1">
                    <nuxt-link to="/">
                        <v-img
                            :src="require('~/assets/images/biogridace_spade_small.png')"
                            contain
                            position="center center"
                        />
                    </nuxt-link>
                </v-list-item-title>
                <v-list-item-title v-else class="title ml-n2">
                    <nuxt-link to="/">
                        <v-img
                            :src="require('~/assets/images/biogridace_logo_small.png')"
                            contain
                            position="center center"
                        />
                    </nuxt-link>
                </v-list-item-title>
            </v-list-item>
        </v-navigation-drawer>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'nuxt-property-decorator'

@Component
export default class TheNavDrawer extends Vue {
    private drawerState: boolean = true;
    private clipped: boolean = false;
    private miniVariant: boolean = false;
    private title: string = process.env.SHORT_TITLE || 'BioGRID';

    private created () {
        this.$nuxt.$on('toggleDrawerVariant', this.toggleDrawerVariant)
    }

    private toggleDrawerVariant () {
        this.miniVariant = !this.miniVariant
    }
}
</script>
