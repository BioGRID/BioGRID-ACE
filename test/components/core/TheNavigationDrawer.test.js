// LIBRARIES
import Vuetify from 'vuetify'
import Vuex from 'vuex'

// COMPONENTS
import TheNavigationDrawer from '@/components/core/TheNavigationDrawer.vue'

// UTILITIES
import { shallowMount, createLocalVue } from '@vue/test-utils'

// SETUP LOCALIZED VUE
const localVue = createLocalVue()
localVue.use(Vuex)

// SETUP MOCKS
const $nuxt = { $on: jest.fn() }
const mocks = {
    $nuxt
}

let wrapper = null

// RUN BEFORE EACH TEST
beforeEach(() => {
    const vuetify = new Vuetify()
    wrapper = shallowMount(TheNavigationDrawer, {
        mocks,
        localVue,
        vuetify
    })
})

// RUN AFTER EACH TEST
afterEach(() => {
    wrapper.destroy()
})

describe('TheNavigationDrawer', () => {
    test('renders', () => {
        expect(wrapper.exists()).toBe(true)
    })
    test('renders properly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
    test('when created, listener is set for event bus', () => {
        expect($nuxt.$on).toHaveBeenCalledWith('toggleDrawerVariant', wrapper.vm.toggleDrawerVariant)
    })
    test('toggle drawer variant', () => {
        expect(wrapper.vm.miniVariant).toBeFalsy()
        wrapper.vm.toggleDrawerVariant()
        expect(wrapper.vm.miniVariant).toBeTruthy()
    })
})
