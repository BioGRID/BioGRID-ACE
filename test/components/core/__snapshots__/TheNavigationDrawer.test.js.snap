// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheNavigationDrawer renders properly 1`] = `
"<div class=\\"the-navigation-drawer\\">
  <nav class=\\"v-navigation-drawer v-navigation-drawer--fixed v-navigation-drawer--is-mobile v-navigation-drawer--open theme--light primary\\" style=\\"height: 100vh; top: 0px; transform: translateX(0%); width: 200px;\\">
    <div class=\\"v-navigation-drawer__content\\">
      <div tabindex=\\"-1\\" class=\\"mt-1 v-list-item theme--light\\">
        <div class=\\"v-list-item__title title ml-n2\\"><a to=\\"/\\">
            <div class=\\"v-image v-responsive theme--light\\">
              <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
                <div class=\\"v-image__image v-image__image--preload v-image__image--contain\\" style=\\"background-position: center center;\\"></div>
              </transition-stub>
              <div class=\\"v-responsive__content\\"></div>
            </div>
          </a></div>
      </div>
    </div>
    <div class=\\"v-navigation-drawer__border\\"></div>
  </nav>
</div>"
`;
